src/main/groovy/life/qbic/utils/BioinformaticAnalysisParser.groovy

```diff
@@ -0,0 +1,369 @@
+import groovy.util.logging.Log4j2
+import life.qbic.datamodel.datasets.NfCorePipelineResult
+import life.qbic.datasets.parsers.DataParserException
+import life.qbic.datasets.parsers.DatasetParser
+import life.qbic.datasets.parsers.DatasetValidationException
+
+import java.nio.file.NotDirectoryException
+import java.nio.file.Path
+import java.text.ParseException
+
+
+/**
+ * <h1>Parser storing the fileTree of a nf-core pipeline output directory into JSON format</h1>
+ * <br>
+ * <p>Converts a fileTree generated by a nf-core pipeline into a JSON String similar to {@link NanoporeParser}
+ *
+ * @since 1.8.0
+ * @param directory path of nf-core directory whose fileTree should be converted into a JSON String
+ *
+ */
+
+@Log4j2
+class BioinformaticAnalysisParser implements DatasetParser<NfCorePipelineResult>{
+
+    /**
+     * Contains the associated keys of the required root directory subFolders
+     *
+     * This enum contains the keys which are associated with the required subFolders of the root directory in the provided fileTree
+     * @since 1.8.0
+     */
+    enum RequiredRootFolderKeys {
+        QUALITYCONTROL("qualityControl"),
+        PIPELINEINFORMATION("pipelineInformation"),
+        PROCESSFOLDERS("processFolders")
+
+        private String keyName
+
+        RequiredRootFolderKeys(String keyName) {
+            this.keyName = keyName
+        }
+
+        String getKeyName() {
+            return this.keyName
+        }
+    }
+
+    /**
+     * Contains the associated keys of the required files in the root directory
+     *
+     * This enum contains the keys which are associated with the required files of the root directory in the provided fileTree
+     * @since 1.8.0
+     */
+    enum RequiredRootFileKeys {
+        RUNID("runId"),
+        SAMPLEID("sampleIds"),
+
+        private String keyName
+
+        RequiredRootFileKeys(String keyName) {
+            this.keyName = keyName
+        }
+
+        String getKeyName() {
+            return this.keyName
+        }
+    }
+
+    /**
+     * Contains the associated keys of the required files in the pipe_line directory
+     *
+     * This enum contains the keys which are associated with the required files in the pipeline directory of the provided fileTree
+     * @since 1.8.0
+     */
+    enum RequiredPipelineFileKeys {
+        SOFTWAREVERSIONS("softwareVersions"),
+        EXECUTIONREPORT("executionReport"),
+        PIPELINEREPORT("pipelineReport")
+
+        private String keyName
+
+        RequiredPipelineFileKeys(String keyName) {
+            this.keyName = keyName
+        }
+
+        String getKeyName() {
+            return this.keyName
+        }
+
+    }
+
+
+    /** {@InheritDoc} */
+    @Override
+    NfCorePipelineResult parseFrom(Path root) throws DataParserException, DatasetValidationException {
+        Map fileTreeMap = parseFileStructureToMap(root)
+        adaptMapToDatasetStructure(fileTreeMap)
+        //TODO validate json with schema
```

Comment:
I consider this part of this PR. The parser is ready when the validation was successful.

Reply:
I'll open a seperate Branch and merge it into this one so it can be tracked seperately 👍

Reply:
It is done!